src/NServiceBus.Core/Diagnostics/Metrics/MessagingMetricsFeature.cs

```diff
@@ -12,12 +12,24 @@ class MessagingMetricsFeature : Feature
             "NServiceBus.Core",
             "0.1.0");
 
-        public MessagingMetricsFeature() => Prerequisite(c => !c.Receiving.IsSendOnlyEndpoint, "Processing metrics are not supported on send-only endpoints");
+        internal static readonly Counter<long> TotalProcessedSuccessfully =
+            NServiceBusMeter.CreateCounter<long>("nservicebus.messaging.successes", description: "Total number of messages processed successfully by the endpoint.");
 
-        /// <inheritdoc />
-        protected internal override void Setup(FeatureConfigurationContext context) => RegisterBehavior(context);
+        internal static readonly Counter<long> TotalFetched =
+            NServiceBusMeter.CreateCounter<long>("nservicebus.messaging.fetches", description: "Total number of messages fetched from the queue by the endpoint.");
+
+        internal static readonly Counter<long> TotalFailures =
+            NServiceBusMeter.CreateCounter<long>("nservicebus.messaging.failures", description: "Total number of messages processed unsuccessfully by the endpoint.");
+
+        public MessagingMetricsFeature()
+        {
+            EnableByDefault();
+            Prerequisite(c => TotalProcessedSuccessfully.Enabled || TotalFetched.Enabled || TotalFailures.Enabled, "No subscribers for messaging metrics");
```

Comment:
Maybe extract as a method/property with a good name, e.g. `HasMetricListner`? Having more counters in the future means this will grow.